TorchSharp/TorchFiles.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+using System.Text;
+
+namespace Torch.IO
+{
+    /// <summary>
+    ///   Abstract base class for all Torch files.
+    /// </summary>
+    public abstract partial class File : IDisposable
+    {
+        internal sealed class HType : SafeHandle
+        {
+            public HType(IntPtr preexistingHandle, bool ownsHandle) : base(IntPtr.Zero, ownsHandle)
+            {
+                SetHandle(preexistingHandle);
+            }
+
+            public override bool IsInvalid => handle == (IntPtr)0;
```

Comment:
Doc please on public methods / properties.

Reply:
The whole HType class is internal. Do we need to have doc comments for those?